src/pages/Dashboard/components/TodosByGoal/index.tsx

```diff
@@ -2,12 +2,18 @@ import { Goal } from '@/types/interface';
 import { FlagIcon } from '@assets';
 import LoadingAnimation from '@components/LoadingAnimation';
 import useGetGoals from '@hooks/api/goalsAPI/useGetGoals';
+import InfiniteScroll from 'react-infinite-scroller';
 import TodosByGoalBox from './TodosByGoalBox';
 
 function TodosByGoal() {
-  const { data, isLoading } = useGetGoals(3);
-  const goalsData = data?.pages[0].data;
-
+  const {
+    data: goalsInfo,
+    isLoading,
+    fetchNextPage,
+    hasNextPage,
+  } = useGetGoals();
+  const goalsData = goalsInfo?.pages || [];
+  const { totalCount } = goalsData[0]?.data || 0;
```

Comment:
```suggestion
  const totalCount = goalsData[0]?.data?.totalCount ?? 0;
```
[Suggestion] 지금 방식은 데이터가 없을 때 구조분해가 안될 것 같아요?!

Reply:
data가 없으면 undefined라서 괜찮지 않을까요

Reply:
0이 구조분해가 안되니까 그럼 이런식으로 써야할걸요??
```suggestion
  const { totalCount } = goalsData[0]?.data || {totalCount: 0} ;
```